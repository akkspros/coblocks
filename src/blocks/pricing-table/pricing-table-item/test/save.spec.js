/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with content', () => {
		block.attributes.title = 'Plan Title';
		block.attributes.amount = '49';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Plan Title' );
		expect( serializedBlock ).toContain( '49' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should not render without content', () => {
		serializedBlock = serialize( createBlock( name ) );
		expect( serializedBlock ).toEqual( `<!-- wp:${ name } /-->` );
	} );

	it( 'should render with content', () => {
		block.attributes.title = 'Plan Title';
		block.attributes.features = '- Feature 1';
		block.attributes.currency = '$';
		block.attributes.amount = '49';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Plan Title' );
		expect( serializedBlock ).toContain( '- Feature 1' );
		expect( serializedBlock ).toContain( '$' );
		expect( serializedBlock ).toContain( '49' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
