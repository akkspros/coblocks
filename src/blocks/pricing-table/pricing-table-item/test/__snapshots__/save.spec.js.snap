// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/pricing-table-item should render with content 1`] = `
"<!-- wp:coblocks/pricing-table-item -->
<div class=\\"wp-block-coblocks-pricing-table-item\\"><span class=\\"wp-block-coblocks-pricing-table-item__title\\">Plan Title</span><div class=\\"wp-block-coblocks-pricing-table-item__price-wrapper\\"><span class=\\"wp-block-coblocks-pricing-table-item__amount\\">49</span></div></div>
<!-- /wp:coblocks/pricing-table-item -->"
`;

exports[`coblocks/pricing-table-item should render with content 2`] = `
"<!-- wp:coblocks/pricing-table-item -->
<div class=\\"wp-block-coblocks-pricing-table-item\\"><span class=\\"wp-block-coblocks-pricing-table-item__title\\">Plan Title</span><div class=\\"wp-block-coblocks-pricing-table-item__price-wrapper\\"><span class=\\"wp-block-coblocks-pricing-table-item__currency\\">$</span><span class=\\"wp-block-coblocks-pricing-table-item__amount\\">49</span></div><ul class=\\"wp-block-coblocks-pricing-table-item__features\\">- Feature 1</ul></div>
<!-- /wp:coblocks/pricing-table-item -->"
`;
